src/objects/ZShape.js

```diff
@@ -0,0 +1,316 @@
+/**
+ * Based on ZSprite by Matevz.
+ *
+ */
+
+import {Mesh} from './Mesh.js';
+import {Quad} from './Quad.js';
+import {ZShapeBasicMaterial} from '../materials/ZShapeBasicMaterial.js';
+import {Vector2} from '../RenderCore.js';
+import {Geometry} from './Geometry.js';
+import {Float32Attribute, Uint32Attribute} from '../core/BufferAttribute.js';
+
+
+export class ZShape extends Mesh {
+    constructor(geometry = null, material = null) {
+        if (geometry === null) {
+            /*
+            let xy0 = new Vector2(-0.5, 0.5);
+            let xy1 = new Vector2(0.5, -0.5);
+            geometry = Quad.makeGeometry(xy0, xy1, false, true, false, false);
+            */
+           geometry = ZShape.makeCubeGeometry();
+           console.log("zshape cube geometry ", geometry);
+           this._geometry = geometry;
```

Comment:
I think you should remove line 24. Accessing "this" is only allowed after calling the super constructor.
The Mesh class should set "this._geometry = geometry" when calling the super constructor (line 37).